streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -276,6 +276,11 @@ private void addTask(final Task task) {
                         changelogReader.transitToUpdateStandby();
                     }
                 }
+
+                // Move task to paused tasks immediately to not make any progress
+                if (topologyMetadata.isPaused(task.id().topologyName())) {
+                    pauseTask(task);
```

Comment:
Why not have this `if` on line 264 and adding it to the `pausedTasks` right away instead of calling `pauseTask()` which does a couple of steps like writing a checkpoint which is not needed at when the task is just added to the state updater?

Reply:
Done. But I also had to make more changes to make resuming reliable. I have to wake up the state updater thread when a topology is resumed, to make sure that the state updater does in fact resume the task